src/frontend/src/optimizer/plan_node/utils.rs

```diff
@@ -377,17 +377,21 @@ pub fn infer_kv_log_store_table_catalog_inner(
 
 /// Check that all leaf nodes must be stream table scan,
 /// since that plan node maps to `backfill` executor, which supports recovery.
-pub(crate) fn plan_has_backfill_leaf_nodes(plan: &PlanRef) -> bool {
+/// Some other leaf nodes like `StreamValues` do not support recovery, and they
+/// cannot use background ddl.
```

Comment:
Do we have an issue for this? Is it possible to address this by simply recording the state into a state table?